### PR TITLE
Brinqa: Initial Commit - First Feature

### PR DESCRIPTION
**Pull request by Jth155**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [x] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [x] If this is a new connector or you are adding new actions
    - [x] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- The query in this app is a query against the Brinqa GraphQL console. There are two direct usages, the first is to put in 
    a singular model name (ie: Asset) to return the attributes that exist under that model. Otherwise, a pluralized version
    of the model is used (ie: assets, note case sensitivity) in conjunction with a filter string (ie: name = *assetName*) 
    and a list of return values, space delineated (ie: ip id name risk) to return information, in the case of the example,
    on an asset with a specific name

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by Jth155** [comment]
Hello, going on 6 weeks here without an update or an acknowledgement. Is there anyway I could get some information on this?

**Comment by ishans-crest** [comment]
Hi @Jth155 
Thanks for your patience. We'll start with the review.

**Comment by Jth155** [comment]
Thank you so much!

**Comment by dhwanis-crest** [comment]
Hi @Jth155,
We have made some code changes as per our standard practice. 
Moreover, there are below mentioned changes that need to be done from your end.

- Our current supported min_phantom_version is "6.0.0". Can you please upgrade the min_phantom_version and test the app on that version?
- There are occurrences of "degub_print" in the connector file which is undefined. The expected method for logging is "debug_print". This needs to be changed
- The app "description" field in the json file contains just the name of the app. It should contain at least 22 letters.

Can you please make the required code changes? Please let us know if you have any concerns.

**Comment by Jth155** [comment]
I'm looking into these now, thanks!

**Comment by Jth155** [comment]
@dhwanis-crest I think I have fixed the issues that you mentioned, let me know if there are any further issues I need to address.

**Comment by dhwanis-crest** [comment]
Hi @Jth155, The product_name is "GraphQL API" which is generic. Should we change it to "Brinqa GraphQL API" or something? Also, we can update the app description according to that.

**Review by mishalp-crest — APPROVED** [review]

**Review by mishalp-crest — APPROVED** [review]

**Review by achandya-crest — APPROVED** [review]

File: brinqa_connector.py [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+            token_type = response["token_type"]
+
+        self.debug_print("Connecting to endpoint")
+        # make a graph rest call to receive information about all items in identifier
+        headers = {"Authorization": token_type + " " + token}
+        if "filter" in param and "return_values" in param:
+            data_model = f"{param['data_model']}"
+            data_model_lower = data_model[0].lower() + data_model[1:]
+            filter_string = f"{param['filter']}"
+            return_string = f"{{{param['return_values']}}}"
```
Comment (alexa-phantom) [inline brinqa_connector.py:284]: What is the reasoning for additional brackets here?

Comment (dhwanis-crest) [reply brinqa_connector.py:284]: The 3 brackets would create a string surrounded with {}. @Jth155, Can you please confirm whether it is the actual use-case or not?

Comment (dhwanis-crest) [reply brinqa_connector.py:284]: Hi @Jth155 , Can you please take a look into this?

Comment (Jth155) [reply brinqa_connector.py:284]: Yes, that is the intended use case. The return values are a bracketed set of attributes that are space delineated at the end of the query.

Comment (dhwanis-crest) [reply brinqa_connector.py:284]: @alexa-phantom , Can you please re-review this?

**Review by alexa-phantom — APPROVED** [review]

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+beautifulsoup4==4.9.1
+requests==2.31.0
```
Comment (rgil-splunk) [inline requirements.txt]: per the most recent discussion https://splunk.slack.com/archives/GBSGHUUD9/p1685364655311529 we should stop explicitly including requests in requirements.txt. this version of requests doesn't work on the most recent cloud release anyways

**Review by ricardo-phantom — APPROVED** [review]

File: requirements.txt (hunk 2 above)
Comment (dhwanis-crest) [reply requirements.txt]: Done with the changes

**Pull request merged**